gravitational/next
# markdown-lint settings update

As a part of resolving https://github.com/gravitational/teleport/issues/10107 I'm updating some rules and settings to resolve issues that happen while integration was broken.

Changes:

- [x] Changed plugin order - includes and variables should be resolved before testing links – fixes false errors with variables in links.
- [x] Added `--silent` flag to the command - info about passed files removed from the CLI.
- [x]  Reduced concurrency for link validation to reduce problem with Github rate limits.
- [x] Update whitelist for links check to exclude some urls.

How to test:

- Check out the branch.
- Run `git submodule update` (resolve updating conflicts if necessary).
- Run `yarn && yarn build-node` (we need to build remark plugins before using them in linter).
- Run `yarn markdown-lint` to test basic errors.
- Run `yarn markdown-lint-external-links` to test basic errors + external links.

First command should return 3 errors for variable names and 6 warnings about `./database-access/faq` path.

Second command should return 103 additional warnings for the broken links.
